src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -138,8 +147,9 @@ public void EmitToolsVersionAttributeToInMemoryProject9()
                 ";
 
             SolutionFile solution = SolutionFile_Tests.ParseSolutionHelper(solutionFileContents);
+            var buildEventContext = new BuildEventContext(0, 0, BuildEventContext.InvalidProjectContextId, 0);
```

Comment:
Wondering if anyone else has more insight into ```BuildEventContext```. All the changed tests were failing unless the third argument to the constructor is ```BuildEventContext.InvalidProjectContextId```, and failing by Windows bringing up a crash window without a useful stack trace. Noticed this because all the passing tests happened to be passing ```BuildEventContext.Invalid``` to ```Generate()```.

Reply:
Could be this check which is executed only when ProjectContextId is valid: https://github.com/Microsoft/msbuild/blob/2a012e653766eb261b09b29b3106f4eb57a7f61d/src/Build/BackEnd/Components/Logging/LoggingService.cs#L1457

You should look at the stack trace and see where the exception is coming from, as it could point to further issues. As far as I can remember, when in Debug, MSBuild sometimes emits that crash window to let you easily attach a debugger. If you click Retry (or abort, can't remember), it will spawn the debugger attach window.